parsl/executors/high_throughput/interchange.py

```diff
@@ -485,8 +489,8 @@ def process_tasks_to_send(self, interesting_managers: Set[bytes]) -> None:
             interesting=len(interesting_managers)))
 
         if interesting_managers and not self.pending_task_queue.empty():
-            shuffled_managers = list(interesting_managers)
-            random.shuffle(shuffled_managers)
+            unsorted_managers = list(interesting_managers)
```

Comment:
you could maybe avoid doing (effectively) a copy here and then another copy inside the manager by defining the interface to be that sort_managers will never modify its input - eg make the unsorted_managers parameter be labelled as `Iterable[bytes]` so type checking will push back on implementations that try to do more than just iterate over it, and change the copy of the list inside the random implementation to make a list from that iterable. (which would actually be a set in practice here)

This probably won't affect performance much in many cases, but with a thousand managers in the list it might be relevant.